packages/fx-core/src/plugins/solution/fx-solution/debug/util/ngrok.ts

```diff
@@ -23,7 +23,7 @@ function delay(ms: number) {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
-export async function getNgrokHttpUrl(port: string | number): Promise<string | undefined> {
+export async function getNgrokHttpUrl(addr: string | number): Promise<string | undefined> {
```

Comment:
In "Set up bot" task, we do not know the bot port, how to get the ngrok url?

Reply:
Please use the static method LocalTunnelTaskTerminal.getNgrokEndpoint()

Reply:
Sure. Just curious how this way will work for CLI or even VS in the future. Do you think it is necessary to have an implementation in fx-core to manage the tunnel, something like what @a1exwang did before.

Reply:
Do we need to maintain ngrok command in cli? We don't need to maintain lifecycle for ngrok of cli. User can directly run the native ngrok command.

Reply:
I believe for CLI and VS we will need to maintain ngrok lifecycle in these cases:
- VS (teamsfx project) - one click F5
- CLI (teamsfx project) - one command preview
- CLI (existing app) - something like `teamsfx setup bot --tunnel` to automatically start tunneling and register bot

Anyway, these are coming features, so for now just ensure there will be no breaking change.